src/components/MultiStepForm/PlanSelection/PlanSelection.tsx

```diff
@@ -0,0 +1,111 @@
+import { Control, UseFormSetValue, Controller, FieldErrors } from 'react-hook-form';
+import { useState, useEffect } from 'react';
+
+import Switch from '@mui/material/Switch';
+import Typography from '@mui/material/Typography';
+
+import { IFormInput } from '../IFormInput';
+import { monthlyPricingPlans } from './Data/PricingPlansData';
+
+import { SelectPlanButton, PlanSelectionContainer, MonthlyYearlyContainer, OfferParagraph, OfferHeader, OfferExtra } from '../PlanSelection/PlanSelectionStyles';
+import { ErrorsLabel, ErrorContainer } from '../PersonalInformation/PersonalInformationStyles';
+
+interface PlanSelectionProps {
+  setValue: UseFormSetValue<IFormInput>;
+  control: Control<IFormInput>;
+  setYearlyPlan: () => void;
+  isYearlyPlan: boolean;
+  errors: FieldErrors<IFormInput>;
+}
+
+const PlanSelection: React.FC<PlanSelectionProps> = ({ control, setValue, setYearlyPlan, isYearlyPlan, errors }) => {
+  const [activeButtonId, setActiveButtonId] = useState<number | null>(null);
+
+  const yearlyPricingPlans = monthlyPricingPlans.map((plan) => ({
+    ...plan,
+    price: plan.price * 10,
```

Comment:
Consider not calculating the pricing at all, rather place this value directly in mocked object